python/gilded_rose.py

```diff
@@ -7,33 +7,56 @@ def __init__(self, items):
 
     def update_quality(self):
         for item in self.items:
-            if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert":
-                if item.quality > 0:
-                    if item.name != "Sulfuras, Hand of Ragnaros":
-                        item.quality = item.quality - 1
-            else:
-                if item.quality < 50:
-                    item.quality = item.quality + 1
-                    if item.name == "Backstage passes to a TAFKAL80ETC concert":
-                        if item.sell_in < 11:
-                            if item.quality < 50:
-                                item.quality = item.quality + 1
-                        if item.sell_in < 6:
-                            if item.quality < 50:
-                                item.quality = item.quality + 1
-            if item.name != "Sulfuras, Hand of Ragnaros":
-                item.sell_in = item.sell_in - 1
-            if item.sell_in < 0:
-                if item.name != "Aged Brie":
-                    if item.name != "Backstage passes to a TAFKAL80ETC concert":
-                        if item.quality > 0:
-                            if item.name != "Sulfuras, Hand of Ragnaros":
-                                item.quality = item.quality - 1
-                    else:
-                        item.quality = item.quality - item.quality
-                else:
+            match item.name:
+                case "Aged Brie":
                     if item.quality < 50:
-                        item.quality = item.quality + 1
+                        item.quality += 1
+                case "Sulfuras, Hand of Ragnaros":
+                    continue
+                case item.name if "Backstage passes" in item.name:
+                    if item.sell_in > 0:
+                        item.quality += 3
+                    else:
+                        item.quality += 2
+                case item.name if "Conjured" in item.name:
+                    if item.sell_in > 0:
+                        item.quality = max(0, item.sell_in - 2)
+                    else:
+                        item.quality = max(0, item.sell_in - 4)
+                case _:
+                    if item.sell_in > 0:
+                        item.quality = max(0, item.sell_in - 1)
+                    else:
+                        item.quality = max(0, item.sell_in - 2)
```

Comment:
I think we can commonise this piece of logic considering that `2 = 2 * 1` and `4 = 2 * 2`.